indico/modules/rb/controllers/backend/bookings.py

```diff
@@ -422,6 +423,31 @@ def _process(self):
         return jsonify(occurrence=reservation_occurrences_schema.dump(self.occurrence, many=False))
 
 
+class RHBookingOccurrenceLinkActions(RHBookingBase):
+    """Links a Reservation occurrence to an event."""
+
+    def _process_args(self):
+        RHBookingBase._process_args(self)
+        date = dateutil.parser.parse(request.view_args['date'], yearfirst=True).date()
```

Comment:
In new code I'd use marshmallow for this:

```py
    @use_kwargs({
        'date': fields.Date(required=True),
    })
    def _process_args(self, date):
```

Reply:
This raises an error. It seems it the parameter is in the path it does not like it.

<img width="442" alt="Screenshot 2024-03-22 at 10 43 38" src="https://github.com/indico/indico/assets/2722756/d420e382-384b-412d-9335-bf5e0d89c3d9">

Reply:
Ah, missed the part of it being in `view_args`. Add `location='view_args'` to the `use_kwargs()` call and it will work.

Reply:
Now it's OK. Thanks!